### PR TITLE
Cert verification for non-CA certs

### PR DESCRIPTION
**Pull request by vishalnayak**
Fixes https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/vault-tool/uVdhJ7DEbYM/pci7UWq-BQAJ

### CODE REVIEW

**Comment by jefferai** [comment]
Thinking back on this I'm not super convinced this wasn't on purpose. The idea being if you are putting in a specific cert then you're really tying to a key more than a lifecycle. I agree that it's _unexpected_ though so probably we should change this.

**Comment by ikatson** [comment]
That error does not seem related, seems like smth else in master is broken

```
command/generate-root_test.go:85: ui.OutputWriter.Bytes undefined (type *cli.syncBuffer has no field or method Bytes)
command/rekey_test.go:185: ui.OutputWriter.Bytes undefined (type *cli.syncBuffer has no field or method Bytes)
command/write_test.go:246: ui.OutputWriter.Bytes undefined (type *cli.syncBuffer has no field or method Bytes)
command/write_test.go:247: ui.OutputWriter.Bytes undefined (type *cli.syncBuffer has no field or method Bytes)
```

File: builtin/credential/cert/path_login.go [hunk 1]
```diff
@@ -175,6 +175,15 @@ func (b *backend) verifyCredentials(req *logical.Request, d *framework.FieldData
 			if tCert.SerialNumber.Cmp(clientCert.SerialNumber) == 0 &&
 				bytes.Equal(tCert.AuthorityKeyId, clientCert.AuthorityKeyId) &&
 				b.matchesConstraints(clientCert, trustedNonCA.Certificates, trustedNonCA) {
+
+				// We are not looking for trusted chains here since this is a
+				// non-CA cert. But validating the connection state detects
+				// expired certificates.
+				_, err := validateConnState(roots, connState)
```
Comment (chrishoffman) [inline builtin/credential/cert/path_login.go]: I think we can just move this check that is happening after this block above the block and avoid the code duplication.

Comment (vishalnayak) [inline builtin/credential/cert/path_login.go]: Done!

**Pull request merged**